### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Related: #4569 #4216 
Fixes: #4571 

`spaces_around_operators` and `spaces_around_brackets` not available yet.

Comment from @pglewis 
Maybe remove the tab size options. No problem if the user has their own preferences. **Only issue it _could_ cause is when dev's are using precision alignment with spaces.**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties
+
+# WordPress Coding Standards
+# http://make.wordpress.org/core/handbook/coding-standards/
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
```
Comment (lougreenwood) [inline .editorconfig]: From the editorconfig docs:

>end_of_line
> Line ending file format (Unix, DOS, Mac)
>The values are case insensitive. They will be lowercased by the core library.

>NOTE: if you want to use native line endings between different operating systems it is better not to set this option and leave that task to the VCS! In the future we might add a value like native for this scenario (cf #226).

So maybe should be omitted?

Comment (JoryHogeveen) [reply .editorconfig]: For PHP it shouldn't matter but since v2.7 is will be required for JS because of the DFV structure templates templates. (@pglewis)

**Comment by pglewis** [comment]
> Comment from @pglewis
> Maybe remove the tab size options. No problem if the user has their own preferences. Only issue it could cause is when dev's are using precision alignment with spaces.

Alignment with spaces should not be affected by the tab size setting, thus why the WordPress whitespace rules use tabs _only_ for indentation at the start of a line.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+indent_size = 4
+tab_width = 4
+insert_final_newline = true
+trim_trailing_whitespace = true
+# spaces_around_operators = true
+# spaces_around_brackets = both
+
+[*.{txt,md}]
+trim_trailing_whitespace = false
+
```
Comment (pglewis) [inline .editorconfig]: Maybe we should consider a section for JSON as well (JSON specs are 2 space indent like YAML)

**Comment by pglewis** [comment]
Also, technically YAML will detect whatever indentation is present and 2 space indent is only a suggestion.  I think I'd rather go ahead and normalize to 2 spaces anyway but both the `.scrutinizer.yml` and `.travis.yml` would need normalized (looks like we're using 4 spaces in each)

**Pull request merged**